lightning/src/ln/channel.rs

```diff
@@ -3466,18 +3459,23 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
 	pub fn get_channel_reestablish(&self) -> msgs::ChannelReestablish {
 		assert_eq!(self.channel_state & ChannelState::PeerDisconnected as u32, ChannelState::PeerDisconnected as u32);
 		assert_ne!(self.cur_remote_commitment_transaction_number, INITIAL_COMMITMENT_NUMBER);
+		// fuzztarget mode marks a subset of pubkeys as invalid so that we can hit "invalid pubkey"
+		// branches, but we unwrap it below, so we arbitrarily select a dummy pubkey which is both
+		// valid, and valid in fuzztarget mode's arbitrary validity criteria:
+		let mut pk = [2; 33]; pk[1] = 0xff;
+		let dummy_pubkey = PublicKey::from_slice(&pk).unwrap();
 		let data_loss_protect = if self.cur_remote_commitment_transaction_number + 1 < INITIAL_COMMITMENT_NUMBER {
 			let remote_last_secret = self.commitment_secrets.get_secret(self.cur_remote_commitment_transaction_number + 2).unwrap();
 			log_trace!(self, "Enough info to generate a Data Loss Protect with per_commitment_secret {}", log_bytes!(remote_last_secret));
 			OptionalField::Present(DataLossProtect {
 				your_last_per_commitment_secret: remote_last_secret,
-				my_current_per_commitment_point: PublicKey::from_secret_key(&self.secp_ctx, &self.build_local_commitment_secret(self.cur_local_commitment_transaction_number + 1))
+				my_current_per_commitment_point: dummy_pubkey
```

Comment:
Can you make a reference here or above to spec, it's pretty confusing to see a dummy key in non-test path. 

Like "if option_static_remotekey applies to the commitment transaction MUST set my_current_commitment_point to a valid point"